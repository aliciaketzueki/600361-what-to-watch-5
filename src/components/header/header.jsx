import React from "react";
import PropTypes from "prop-types";
import {Link} from "react-router-dom";

const Header = (props) => {
  const {header, history} = props;
  const {title, headClass, nav, login} = header;

  const onLoginClick = () => {
    history.push(`/login`);
  };

  return (
    <header className={`page-header ${headClass}`}>
      <div className="logo">
        <Link to="/" className="logo__link">
          <span className="logo__letter logo__letter--1">W</span>
          <span className="logo__letter logo__letter--2">T</span>
          <span className="logo__letter logo__letter--3">W</span>
        </Link>
      </div>

      {title ? <h1 className="page-title user-page__title">{title}</h1> : ``}

      {nav ?
        <nav className="breadcrumbs">
          <ul className="breadcrumbs__list">
            <li className="breadcrumbs__item">
              <a href="movie-page.html" className="breadcrumbs__link">The Grand Budapest Hotel</a>
            </li>
            <li className="breadcrumbs__item">
              <a className="breadcrumbs__link">Add review</a>
            </li>
          </ul>
        </nav>
        : ``
      }

      {login ?
        <div className="user-block">
          <div className="user-block__avatar" onClick={onLoginClick}>
            <img src="img/avatar.jpg" alt="User avatar" width="63" height="63" />
          </div>
        </div>
        : ``
      }

    </header>
  );
};

export default Header;

Header.propTypes = {
  header: PropTypes.shape().isRequired,
  history: PropTypes.shape().isRequired,
};
