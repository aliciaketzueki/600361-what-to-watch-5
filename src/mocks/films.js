export default [{
  name: `Fantastic Beasts: The Crimes of Grindelwald`,
  poster: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  src: `https://docs.google.com/uc?id=1lOjED9R5dp-caVj1S8Wf8jjr3ob_qThW`,
  genre: `comedies`
}, {
  name: `Bohemian Rhapsody`,
  poster: `bohemian-rhapsody.jpg`,
  src: `https://docs.google.com/uc?id=1lOjED9R5dp-caVj1S8Wf8jjr3ob_qThW`,
  genre: `dramas`
}, {
  name: `Macbeth`,
  poster: `macbeth.jpg`,
  src: `https://docs.google.com/uc?id=1lOjED9R5dp-caVj1S8Wf8jjr3ob_qThW`,
  genre: `comedies`
}, {
  name: `Aviator`,
  poster: `aviator.jpg`,
  src: `https://docs.google.com/uc?id=1lOjED9R5dp-caVj1S8Wf8jjr3ob_qThW`,
  genre: `romance`
}, {
  name: `We need to talk about Kevin`,
  poster: `we-need-to-talk-about-kevin.jpg`,
  src: `https://docs.google.com/uc?id=1lOjED9R5dp-caVj1S8Wf8jjr3ob_qThW`,
  genre: `thrillers`
}, {
  name: `What We Do in the Shadows`,
  poster: `what-we-do-in-the-shadows.jpg`,
  src: `https://docs.google.com/uc?id=1lOjED9R5dp-caVj1S8Wf8jjr3ob_qThW`,
  genre: `documentary`
}, {
  name: `Revenant`,
  poster: `revenant.jpg`,
  src: `https://docs.google.com/uc?id=1lOjED9R5dp-caVj1S8Wf8jjr3ob_qThW`,
  genre: `comedies`
}, {
  name: `Johnny English`,
  poster: `johnny-english.jpg`,
  src: `https://docs.google.com/uc?id=1lOjED9R5dp-caVj1S8Wf8jjr3ob_qThW`,
  genre: `kids & family`
}, {
  name: `Shutter Island`,
  poster: `shutter-island.jpg`,
  src: `https://docs.google.com/uc?id=1lOjED9R5dp-caVj1S8Wf8jjr3ob_qThW`,
  genre: `thrillers`
}, {
  name: `Pulp Fiction`,
  poster: `pulp-fiction.jpg`,
  src: `https://docs.google.com/uc?id=1lOjED9R5dp-caVj1S8Wf8jjr3ob_qThW`,
  genre: `crime`
}, {
  name: `No Country for Old Men`,
  poster: `no-country-for-old-men.jpg`,
  src: `https://docs.google.com/uc?id=1lOjED9R5dp-caVj1S8Wf8jjr3ob_qThW`,
  genre: `kids & family`
}, {
  name: `Snatch`,
  poster: `snatch.jpg`,
  src: `https://docs.google.com/uc?id=1lOjED9R5dp-caVj1S8Wf8jjr3ob_qThW`,
  genre: `romance`
}, {
  name: `Moonrise Kingdom`,
  poster: `moonrise-kingdom.jpg`,
  src: `https://docs.google.com/uc?id=1lOjED9R5dp-caVj1S8Wf8jjr3ob_qThW`,
  genre: `horror`
}, {
  name: `Seven Years in Tibet`,
  poster: `seven-years-in-tibet.jpg`,
  src: `https://docs.google.com/uc?id=1lOjED9R5dp-caVj1S8Wf8jjr3ob_qThW`,
  genre: `comedies`
}, {
  name: `Midnight Special`,
  poster: `midnight-special.jpg`,
  src: `https://docs.google.com/uc?id=1lOjED9R5dp-caVj1S8Wf8jjr3ob_qThW`,
  genre: `comedies`
}, {
  name: `War of the Worlds`,
  poster: `war-of-the-worlds.jpg`,
  src: `https://docs.google.com/uc?id=1lOjED9R5dp-caVj1S8Wf8jjr3ob_qThW`,
  genre: `horror`
}, {
  name: `Dardjeeling Limited`,
  poster: `dardjeeling-limited.jpg`,
  src: `https://docs.google.com/uc?id=1lOjED9R5dp-caVj1S8Wf8jjr3ob_qThW`,
  genre: `crime`
}, {
  name: `Orlando`,
  poster: `orlando.jpg`,
  src: `https://docs.google.com/uc?id=1lOjED9R5dp-caVj1S8Wf8jjr3ob_qThW`,
  genre: `sci-fi`
}, {
  name: `Mindhunter`,
  poster: `mindhunter.jpg`,
  src: `https://docs.google.com/uc?id=1lOjED9R5dp-caVj1S8Wf8jjr3ob_qThW`,
  genre: `documentary`
}, {
  name: `Midnight Special`,
  poster: `midnight-special.jpg`,
  src: `https://docs.google.com/uc?id=1lOjED9R5dp-caVj1S8Wf8jjr3ob_qThW`,
  genre: `sci-fi`
}];
