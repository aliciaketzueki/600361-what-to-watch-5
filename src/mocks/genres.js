export default [
  {name: `All genres`},
  {name: `Crime`},
  {name: `Comedies`},
  {name: `Documentary`},
  {name: `Dramas`},
  {name: `Horror`},
  {name: `Kids & Family`},
  {name: `Romance`},
  {name: `Sci-Fi`},
  {name: `Thrillers`}
];
