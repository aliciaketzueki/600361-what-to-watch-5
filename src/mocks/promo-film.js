export default {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  year: 2014,
  genreId: 15,
};
