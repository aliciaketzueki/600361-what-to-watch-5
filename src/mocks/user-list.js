export default [{
  name: `Aviator`,
  poster: `aviator.jpg`,
  video: `https://docs.google.com/uc?id=1lOjED9R5dp-caVj1S8Wf8jjr3ob_qThW`,
  genre: `romance`,
}, {
  name: `Revenant`,
  poster: `revenant.jpg`,
  video: `https://docs.google.com/uc?id=1lOjED9R5dp-caVj1S8Wf8jjr3ob_qThW`,
  genre: `comedies`
}, {
  name: `Johnny English`,
  poster: `johnny-english.jpg`,
  video: `https://docs.google.com/uc?id=1lOjED9R5dp-caVj1S8Wf8jjr3ob_qThW`,
  genre: `kids & family`
}];
