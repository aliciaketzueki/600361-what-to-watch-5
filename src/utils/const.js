export const MORE_LIKE_NUM = 4;
export const MAX_ACTORS_NUM = 4;
export const MAX_GENRES_NUM = 10;
export const SECOND = 1000;
export const SHORT_TIMEOUT = 10;
export const INITIAL_FILMS_NUM = 8;
export const ALL_GENRES = `All genres`;

export const Validation = {
  EMAIL: /^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i,
  PASSWORD: 6,
  COMMENT_MIN: 50,
  COMMENT_MAX: 400
};

export const Errors = {
  EMAIL: `Please enter a valid email address`,
  PASSWORD: `Password must be at least ${Validation.PASSWORD} characters`,
  TEXT_REVIEW: `Review text must be at least ${Validation.COMMENT_MIN} and no more than ${Validation.COMMENT_MAX} characters`
};

export const tabs = [{
  name: `Overview`,
  id: 0,
}, {
  name: `Details`,
  id: 1,
}, {
  name: `Reviews`,
  id: 2
}];

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const AppRoute = {
  ROOT: `/`,
  LOGIN: `/login`,
  MY_LIST: `/my-list`,
  FILM: `/films/:id`,
  ADD_REVIEW: `/films/:id/review`,
  PLAYER: `/player/:id`,
};

export const APIRoute = {
  FILMS: `/films`,
  PROMO: `/films/promo`,
  COMMENTS: `comments`,
  LOGIN: `/login`,
  FAVOURITES: `/favorite`,
  PLAYER: `/player`
};
