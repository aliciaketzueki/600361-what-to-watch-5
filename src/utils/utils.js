import genres from "../mocks/genres";

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

const getRandom = (min, max) => {
  return Math.floor(Math.random() * (max - min) + min);
};

export const createFilmCard = (arr) => {
  const filmArr = [];

  [].forEach.call(arr, (name) => {
    const film = {};
    film.year = getRandom(1990, 2020);
    film.videoSrc = `https://docs.google.com/uc?id=1lOjED9R5dp-caVj1S8Wf8jjr3ob_qThW`;
    film.id = getRandom(1000, 9999);
    film.name = name;
    film.poster = film.name.split(` `).join(`-`).split(/[^a-zA-Z0-9_\-]/).join(``).toLowerCase() + `.jpg`;
    film.genreId = getRandom(0, 9);
    film.genre = genres[film.genreId].name;
    filmArr.push(film);
  });

  return filmArr;
};
