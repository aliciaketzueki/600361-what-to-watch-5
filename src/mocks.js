import React from "react";
import {AuthorizationStatus, AppRoute, ALL_GENRES, INITIAL_FILMS_NUM} from "./utils/const";
import {validChildren} from "./utils/props";

export const FILM = {
  id: 1,
  name: `The Grand Budapest Hotel`,
  posterImage: `img/the-grand-budapest-hotel-poster.jpg`,
  previewImage: `img/the-grand-budapest-hotel.jpg`,
  backgroundImage: `img/the-grand-budapest-hotel-bg.jpg`,
  backgroundColor: `#ffffff`,
  videoLink: `https://some-link`,
  previewVideoLink: `https://some-link`,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
  rating: 8.9,
  scoresCount: 240,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
  runTime: 99,
  genre: `Comedy`,
  released: 2014,
  isFavorite: false
};

export const FILMS = [FILM];

export const FILM_ID = FILM.id;

export const GENRE = `Crime`;

export const GENRES_LIST = [ALL_GENRES, GENRE];

export const FILMS_RENDERED = INITIAL_FILMS_NUM;

export const REVIEW = {
  id: 1,
  user: {
    id: 4,
    name: `Kate Muir`
  },
  rating: 8.9,
  comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
  date: `2019-05-08T14:13:56.569Z`
};

export const REVIEWS = [REVIEW];

export const URL = AppRoute.ROOT;

export const STATUS = AuthorizationStatus.AUTH;

export const AUTH_INFO = {
  id: 1,
  email: `fd@mail.com`,
  name: `fd`,
  avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg`
};

export const USER_DATA = {
  email: `Oliver.conner@gmail.com`,
  password: `12345678`
};

export const MOCK_EMAIL = USER_DATA.email;

export const MOCK_PASSWORD = `12345678796`;

export const NOOP = () => {};

export const COMMENT = `This is a test comment to check input changing`;

export const RATING = 6;

export const HEADER = {
  title: `My list`,
  headClass: `movie-card__head`,
  nav: true
};

export const MOCK_COMPONENT = () => <div />;

export const MOCK_COMPONENT_WITH_CHILDREN = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

export const MOVIE = {
  duration: 1000,
  progress: 10
};

export const ERROR_STATUS = {
  status: 404,
  statusText: `Not Found`
};

MOCK_COMPONENT_WITH_CHILDREN.propTypes = {
  children: validChildren
};
