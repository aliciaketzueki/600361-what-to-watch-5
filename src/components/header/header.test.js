import React from "react";
import renderer from "react-test-renderer";
import {Header} from "./header";
import {BrowserRouter} from "react-router-dom";
import {AuthorizationStatus} from "../../utils/const";
import {FILM, USER_DATA, HEADER, NOOP} from "../../mocks";

describe(`Render Header`, () => {
  test.each([
    [`with`, AuthorizationStatus.AUTH],
    [`without`, AuthorizationStatus.NO_AUTH],
  ])(`%s auth`, (_expected, login) => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <Header
              header={HEADER}
              login={login}
              userData={USER_DATA}
              film={FILM}
              moveToPage={NOOP}
            />
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
