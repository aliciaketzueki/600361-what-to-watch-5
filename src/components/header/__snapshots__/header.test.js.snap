// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Header with auth 1`] = `
<header
  className="page-header movie-card__head"
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <h1
    className="page-title user-page__title"
  >
    My list
  </h1>
  <nav
    className="breadcrumbs"
  >
    <ul
      className="breadcrumbs__list"
    >
      <li
        className="breadcrumbs__item"
      >
        <a
          className="breadcrumbs__link"
          href="/films/1"
          onClick={[Function]}
        >
          The Grand Budapest Hotel
        </a>
      </li>
      <li
        className="breadcrumbs__item"
      >
        <a
          className="breadcrumbs__link"
        >
          Add review
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="user-block"
  >
    <div
      className="user-block__avatar"
      onClick={[Function]}
    >
      <img
        alt="User avatar"
        height="63"
        width="63"
      />
    </div>
  </div>
</header>
`;

exports[`Render Header without auth 1`] = `
<header
  className="page-header movie-card__head"
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <h1
    className="page-title user-page__title"
  >
    My list
  </h1>
  <nav
    className="breadcrumbs"
  >
    <ul
      className="breadcrumbs__list"
    >
      <li
        className="breadcrumbs__item"
      >
        <a
          className="breadcrumbs__link"
          href="/films/1"
          onClick={[Function]}
        >
          The Grand Budapest Hotel
        </a>
      </li>
      <li
        className="breadcrumbs__item"
      >
        <a
          className="breadcrumbs__link"
        >
          Add review
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="user-block"
  >
    <a
      className="user-block__link"
      href="/login"
      onClick={[Function]}
    >
      Sign in
    </a>
  </div>
</header>
`;
